pkg/controllers/deprovisioning/nodeclaim_consolidation_test.go

```diff
@@ -2264,6 +2267,327 @@ var _ = Describe("NodeClaim/Consolidation", func() {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				ExpectReconcileSucceeded(ctx, deprovisioningController, client.ObjectKey{})
+			}()
+
+			// Iterate in a loop until we get to the validation action
+			// Then, apply the pods to the cluster and bind them to the nodes
+			for {
+				time.Sleep(250 * time.Millisecond)
+				if fakeClock.HasWaiters() {
+					break
+				}
+			}
+			doNotEvictPods := test.Pods(2, test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1alpha5.DoNotEvictPodAnnotationKey: "true",
+					},
+				},
+			})
+			ExpectApplied(ctx, env.Client, doNotEvictPods[0], doNotEvictPods[1])
+			ExpectManualBinding(ctx, env.Client, doNotEvictPods[0], node)
+			ExpectManualBinding(ctx, env.Client, doNotEvictPods[1], node2)
+
+			// Step forward to satisfy the validation timeout and wait for the reconcile to finish
+			fakeClock.Step(45 * time.Second)
+			wg.Wait()
+
+			// we would normally be able to consolidate down to a single node, but we are blocked by the do-not-evict pods during validation
+			Expect(ExpectNodeClaims(ctx, env.Client)).To(HaveLen(2))
+			Expect(ExpectNodes(ctx, env.Client)).To(HaveLen(2))
+			ExpectExists(ctx, env.Client, node)
+			ExpectExists(ctx, env.Client, node2)
+		})
+		It("should not delete node if pods schedule with karpenter.sh/do-not-disrupt during the TTL wait", func() {
+			pods := test.Pods(2, test.PodOptions{})
+			ExpectApplied(ctx, env.Client, nodePool, nodeClaim, node, nodeClaim2, node2, pods[0], pods[1])
+
+			// bind pods to node
+			ExpectManualBinding(ctx, env.Client, pods[0], node)
+			ExpectManualBinding(ctx, env.Client, pods[1], node2)
+
+			// inform cluster state about nodes and machines
+			ExpectMakeNodesAndNodeClaimsInitializedAndStateUpdated(ctx, env.Client, nodeStateController, nodeClaimStateController, []*v1.Node{node, node2}, []*v1beta1.NodeClaim{nodeClaim, nodeClaim2})
+
+			fakeClock.Step(10 * time.Minute)
+
+			var wg sync.WaitGroup
+
+			// Trigger the reconcile loop to start but don't trigger the verify action
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				ExpectReconcileSucceeded(ctx, deprovisioningController, client.ObjectKey{})
+			}()
+
+			// Iterate in a loop until we get to the validation action
+			// Then, apply the pods to the cluster and bind them to the nodes
+			for {
+				time.Sleep(250 * time.Millisecond)
+				if fakeClock.HasWaiters() {
+					break
+				}
+			}
+			doNotEvictPods := test.Pods(2, test.PodOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						v1beta1.DoNotDisruptAnnotationKey: "true",
+					},
+				},
+			})
+			ExpectApplied(ctx, env.Client, doNotEvictPods[0], doNotEvictPods[1])
+			ExpectManualBinding(ctx, env.Client, doNotEvictPods[0], node)
+			ExpectManualBinding(ctx, env.Client, doNotEvictPods[1], node2)
+
+			// Step forward to satisfy the validation timeout and wait for the reconcile to finish
+			fakeClock.Step(45 * time.Second)
```

Comment:
Can we use `ExpectTriggerVerifyAction()` for each time you make this call? Will make sure that we're stepping forward as many times as necessary so that it works and we don't have to adjust this when we make this validation period configurable.